internal/server/data/identity.go

```diff
@@ -62,7 +62,10 @@ func AssignIdentityToGroups(db *gorm.DB, user *models.Identity, provider *models
 			}
 		}
 		if !found {
-			group := &models.Group{Name: name}
+			group := &models.Group{
+				Name:              name,
+				CreatedByProvider: true,
```

Comment:
Should we track the provider ID that actually created the group so we can have a full way to trace back to the origin?

Reply:
Yes, I think that makes sense.